api/tools/src/main/java/org/pmiops/workbench/tools/ExportWorkspaceData.java

```diff
@@ -180,22 +183,26 @@ public CommandLineRunner run(
 
       log.info("collecting all users");
       List<WorkspaceExportRow> rows = new ArrayList<>();
-      Set<DbUser> usersWithoutWorkspaces =
-          Streams.stream(userDao.findAll()).collect(Collectors.toSet());
+      Map<Long, DbUser> usersById =
+          Streams.stream(userDao.findAll())
+              .collect(Collectors.toMap(DbUser::getUserId, Function.identity()));
+
+      Set<Long> userIdsWithoutWorkspaces = usersById.keySet();
 
       log.info("collecting / converting all workspaces");
       for (DbWorkspace workspace : this.workspaceDao.findAll()) {
         rows.add(toWorkspaceExportRow(workspace, includeDemographics));
-        usersWithoutWorkspaces.remove(workspace.getCreator());
+        userIdsWithoutWorkspaces.remove(workspace.getCreator().getUserId());
```

Comment:
Ah, I didn't notice this was `Set<DbUser>`, that would explain the behavioral change. Thanks for catching this.